frida_tools/tracer.py

```diff
@@ -918,6 +926,29 @@ def _sync_handlers(self, change_id):
                 self._notify_update(function, new_handler, handler_file)
 
 
+class OutputFile(object):
+    def __init__(self, filename, truncate=False):
+        self.filename = filename
+        self._truncate = truncate
+        self._fd = None
+
+    def __enter__(self):
+        if self.filename:
```

Comment:
This check is not needed.

Reply:
its needed if you call the constructor with filename=None.. which shouldnt happen, but with dynamic languages everything can break at anytime.

Reply:
this is no longer there, early failure in case the file cant be opened is working and the caes to open a file with no name "" is no longer possible now

Reply:
Nice. Even if anything is possible with dynamic languages, I'm a fan of deciding on API contracts that only support exactly what I need and nothing more, and write the code assuming that the contract is followed exactly. (The public API boundary is a different beast though, as users will be making mistake and usability is important, but for internal APIs I try hard to limit options so the code is as readable as possible.)